File: backend/config/geminiClient.js
import { GoogleGenerativeAI } from "@google/generative-ai";
import { env } from "./env.js";

const GEMINI_MODEL = "gemini-1.5-flash";

let genAI = null;
let geminiModel = null;

if (env.geminiApiKey) {
  genAI = new GoogleGenerativeAI(env.geminiApiKey);
  geminiModel = genAI.getGenerativeModel({ model: GEMINI_MODEL });
}

export const isGeminiConfigured = () => Boolean(geminiModel);

export const getGeminiModel = () => geminiModel;

export const GEMINI_MODEL_NAME = GEMINI_MODEL;

File: backend_flask/config/gemini_client.py
import os
import google.generativeai as genai
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Configure Gemini API Key
api_key = os.getenv("GEMINI_API_KEY")
if not api_key:
    raise ValueError("GEMINI_API_KEY not found in environment variables")

genai.configure(api_key=api_key)

# Initialize Gemini model (gemini-2.0-flash)
def get_gemini_model():
    return genai.GenerativeModel("gemini-2.0-flash")
